source/PyMaterialX/PyMaterialXCore/PyDocument.cpp

```diff
@@ -59,7 +71,9 @@ void bindPyDocument(py::module& mod)
         .def("removeTypeDef", &mx::Document::removeTypeDef)
         .def("addNodeDef", &mx::Document::addNodeDef,
             py::arg("name") = mx::EMPTY_STRING, py::arg("type") = mx::DEFAULT_TYPE_STRING, py::arg("node") = mx::EMPTY_STRING)
-        .def("addNodeDefFromGraph", &mx::Document::addNodeDefFromGraph)
```

Comment:
For the deprecated Python method, I'd instead recommend following the pattern here, which leverages the existing deprecation system in the Python language:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/python/MaterialX/main.py#L178

Reply:
Since the original Python function for `addNodeDefFromGraph` required all seven arguments to be stated explicitly (without any support for default arguments), I believe you can just define a deprecated Python function named `addNodeDefFromGraph` with exactly seven arguments, where its implementation simply calls into the new API.

To give a concrete example of the industry usage that we're aiming to support, here's one from the QuiltiX project, and you can see that they're stating all seven arguments as we previously required:

https://github.com/Herizoran/QuiltiX/blob/d6466f954553c622aed1ef58c66911486c361e95/src/QuiltiX/quiltix.py#L814

Reply:
Hi @jstone-lucasfilm ,

Yes this will work with what I have done currently I don't see any way to do it inside of `main.py`
The issue is that Python does not allow operator overloading to be done dynamically AFAIK.
e.g.
```python
class Test:
    def test(self, arg1):
        print(arg1)

instance = Test()
instance.test("Original method")    

def _test(self, arg1, arg2):
    print(arg1, arg2)

Test.test2 = _test
Test.test = _test # This just replaces the original and does not add a new version of test.

instance = Test()
instance.test("1 arg") # This will error out
instance.test("1 arg", "2 arg")
```

Reply:
Ah, I see what you mean now.  Ok, in that case, your workaround seems fine to me!